lib/token-types/bearer-token-type.js

```diff
@@ -0,0 +1,52 @@
+
+/**
+ * Module dependencies.
+ */
+
+var InvalidArgumentError = require('../errors/invalid-argument-error');
+
+/**
+ * Constructor.
+ */
+
+function BearerTokenType(accessToken, accessTokenLifetime, refreshToken, scope) {
+  if (!accessToken) {
+    throw new InvalidArgumentError('Missing parameter: `accessToken`');
+  }
+
+  this.accessToken = accessToken;
+  this.accessTokenLifetime = accessTokenLifetime;
+  this.refreshToken = refreshToken;
+  this.scope = scope;
```

Comment:
I completely support adding optional properties (could be a very useful feature) but I'm afraid it might lead our users to tricky situations. For example, take this model implementation:

```js
var model = {
  saveToken: function() {
    return db.createToken({
      access_token: 'foo',
      refresh_token: 'bar',
      example_parameter: 'example_value',
      secret_key: 'my-secret-key'
    });
  }
```

In this example, the user will also be exposing their "secret_key", potentially without realizing it.

Maybe we could warn users in the documentation that the entire object will be exposed in the response?